model/src/signature.rs

```diff
@@ -0,0 +1,208 @@
+use super::DomainSeparator;
+use primitive_types::{H160, H256};
+use serde::{de, Deserialize, Serialize};
+use std::fmt;
+use web3::{
+    signing::{self, Key, SecretKeyRef},
+    types::Recovery,
+};
+
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "lowercase")]
+pub enum SigningScheme {
+    Eip712,
+    EthSign,
+}
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Default, Deserialize, Serialize, Hash)]
+pub struct Signature(EcdsaSignature);
```

Comment:
I assume this will become an `enum` or something once pre-sign gets introduced?

Reply:
Yep!